utilities/transform_utils/image_cutting.py

```diff
@@ -0,0 +1,449 @@
+    in row major order.
+    """
+
+    im_shape = image.shape
+    if (len(im_shape) != 2) and (len(im_shape) != 3):
+        raise ValueError("input image must be either a matrix, or a 3d tensor")
+
+    if (len(im_shape) == 2) and num_bands != 1:
+        raise ValueError(
+            "input image is a 2d matrix, but input number of bands is not 1"
+        )
+
+    if len(im_shape) == 2:
+        image = np.expand_dims(image, axis=-1)
+
+    edge_space = (im_shape[1] % cut_dims[1], im_shape[0] % cut_dims[0])
+
+    num_ims_x = int(np.ceil(im_shape[1] / cut_dims[1]))
+    num_ims_y = int(np.ceil(im_shape[0] / cut_dims[0]))
+    num_ims = num_ims_x * num_ims_y
+
+    cut_ims = np.zeros((num_ims, cut_dims[0], cut_dims[1], num_bands), dtype=np.int16)
+
+    i = 0
+
+    for y_ims in range(num_ims_y):
+        for x_ims in range(num_ims_x):
+            if (
+                x_ims == num_ims_x - 1
+                and y_ims != num_ims_y - 1
+                and edge_space != (0, 0)
+            ):
+                cut_ims[i, :, : edge_space[1], :] = image[
+                    y_ims * cut_dims[0] : (y_ims + 1) * cut_dims[0],
+                    x_ims * cut_dims[1] :,
+                    :,
+                ]
+            elif (
+                x_ims != num_ims_x - 1
+                and y_ims == num_ims_y - 1
+                and edge_space != (0, 0)
+            ):
+                cut_ims[i, : edge_space[0], :, :] = image[
+                    y_ims * cut_dims[0] :,
+                    x_ims * cut_dims[1] : (x_ims + 1) * cut_dims[1],
+                    :,
+                ]
+            elif (
+                x_ims == num_ims_x - 1
+                and y_ims == num_ims_y - 1
+                and edge_space != (0, 0)
+            ):
+                cut_ims[i, : edge_space[0], : edge_space[1], :] = image[
+                    y_ims * cut_dims[0] :, x_ims * cut_dims[1] :, :
+                ]
+            else:
+                cut_ims[i, :, :, :] = image[
+                    y_ims * cut_dims[0] : (y_ims + 1) * cut_dims[0],
+                    x_ims * cut_dims[1] : (x_ims + 1) * cut_dims[1],
+                    :,
+                ]
+            i += 1
+
+    return cut_ims
+
+
+def image_cut(
+    image: NDArray[Any], cut_dims: tuple[int, int], num_bands: int = 1, pad: bool = True
+) -> NDArray[Any]:
+    """
+    Takes an input image "image" and cuts into many images each of dimensions "cut_dims".
+    Assumed input image shape
+
+    Note, image are cut in row major order.
+
+    Note, if image dimensions arent a multiple of cut_dims, then the input image is padded
+    at the right, and left edge with black pixels. So some images will contain black areas.
+
+    Arguments
+    ---------
+    :NDArray[Any] image: Numpy array representing image to cut [H,W,C].
+    :tuple[int, int] cut_dims: desired dimensions to cut images to.
+    :int, optional num_bands: Number of bands of input image. Defaults to 1.
```

Comment:
really the number of bands in the input images should not a function argument instead check the images to be cut